### PR TITLE
allow contains to use chained subject as priority element

### LINKED ISSUE #1887: `contains()` targets element with `pointer-events: none` and fails to click instead of targeting element higher up the tree
### Is this a Feature or Bug?

~Bug.~ A feature, I guess. Thought it was a bug, but it turns out `contains` does something else than what I originally assumed.

### Current behavior:

I use UI components from Ant Design (React) and the [button](https://ant.design/components/button/) has the following mark-up:

```html
<button type="button" class="ant-btn">
  <span>Some Label</span>
</button>
```

and the following style applied:

```css
.ant-btn > span {
  pointer-events: none;
}
```

When I try to interact with the button using:

```javascript
cy.get('button').contains('Some Label').click()
```

I get the following error message:

> CypressError: Timed out retrying: cy.click() failed because this element:
> 
> <span>Create</span>
> 
> is being covered by another element:
> 
> <button type="button" class="ant-btn">...</button>
> 
> Fix this problem, or use {force: true} to disable error checking.
> 
> https://on.cypress.io/element-cannot-be-interacted-with

When I programmatically set `pointer-events` to `auto`, the issue disappears. Apparently the `contains` operator finds the `<span>` containing the text and then fails, because it has `pointer-events: none`.

### Desired behavior:

The `contains()` should ideally react to the expected interaction request by selecting elements without `pointer-events: none` when possible. In this case, it should select the `button` and the `click` should be triggered on it.

### Steps to reproduce:

https://github.com/hon2a/cypress-test-tiny/tree/hon2a/pointer-events-bug

### Versions

Cypress: 3.0.1
OS: macOS 10.13.5
browser: Chrome 66.0

### PR DESCRIPTION
**Pull request by kuceb**
fix #1887

this PR allows the chained element to be considered for priority:

e.g.: given `button > span > "some text"`, I want to click the button that contains "some text"
`cy.get('button').contains('some text').click()`

Previously, this would not click the button, but the span.
Now, this would click the button

**note:** this is not making `cy.get('.section').contains('text')` equivalent to `cy.contains('.section', 'text')`, since `.section` is not a priority element (`'input[type=\'submit\'], button, a, label'`)

### CODE REVIEW

**Comment by lukeapage** [comment]
I can see how the new behaviour is preferred.. but this is a big breaking change for us.
We've used contains as a kind of pseudo-selector e.g.

```
cy.get('.tst-section-a')
   .contains('Cancel')
   .click();
```

If you didn't want to change the target couldn't you write `.should('contain', 'Cancel')` ?

**Comment by Bkucera** [comment]
@lukeapage as long as the element is not a priority element, nothing would change,
is `.tst-section-a` a `link, a, or button` ?

**Comment by lukeapage** [comment]
No it’s not a button, it’s a section. And the button only contains the text

**Comment by lukeapage** [comment]
Oh, wow ok.. that is okay. It’s pretty non straight forward, hard to predict behavior.

**Comment by Bkucera** [comment]
@lukeapage agreed.

File: packages/driver/test/cypress/integration/commands/querying_spec.coffee [hunk 1]
```diff
@@ -1280,6 +1280,12 @@ describe "src/cy/commands/querying", ->
         expect($el.length).to.eq(1)
         expect($el).to.match("a")
 
+    it "can pick chained element as priority element", ->
+      $btn = cy.$$('button:contains(click button)')
+      console.log($btn)
```
Comment (brian-mann) [inline packages/driver/test/cypress/integration/commands/querying_spec.coffee]: console log :-/

File: packages/driver/src/cy/commands/querying.coffee [hunk 2]
```diff
@@ -370,7 +370,12 @@ module.exports = (Commands, Cypress, cy, state, config) ->
 
         cy.now("get", selector, getOpts).then ($el) ->
           if $el and $el.length
-            $el = $dom.getFirstDeepestElement($el)
+            elArray = [].slice.call($el)
```
Comment (brian-mann) [inline packages/driver/src/cy/commands/querying.coffee]: $el.get()

File: packages/driver/test/cypress/integration/commands/querying_spec.coffee [hunk 3]
```diff
@@ -1218,7 +1218,7 @@ describe "src/cy/commands/querying", ->
         expect($span.get(0)).to.eq span.get(0)
 
     it "GET is scoped to the current subject", ->
-      span = cy.$$("#click-me a span")
+      span = cy.$$("#click-me a")
```
Comment (brian-mann) [inline packages/driver/test/cypress/integration/commands/querying_spec.coffee]: no longer a `span`

why modify this test?

**Review by brian-mann — CHANGES_REQUESTED** [review]

**Comment by jennifer-shehane** [comment]
Unfortunately we have to close this due to inactivity. Please reopen if there is work to be done addressing this again.

**Pull request closed without merging**